### ISSUE TITLE
Procedure reference

### ISSUE THREAD

**Issue by rokf** [PaulBatchelor/runt#2]
Have you thought about adding a procedure reference in the same format you did for **ugens** in `Sporth`?

I've generated a *template* from the `basic.c` file. Could probably fill this in myself but it would take me much longer :smile: as I'd have to look up everything from the sources.

[proc_reference.txt](https://github.com/PaulBatchelor/runt/files/1367281/proc_reference.txt)

The script which generated this is
```Lua

local re = require 're' -- luarocks install lpeg

local sourcef = io.open(arg[1],'r')
local sourcet = sourcef:read('*all')
sourcef:close()

local pat = re.compile([[
names <- {| (name / .)+ |} -> transform
name <- 'runt_keyword_define' [^"]+ '"' { [^"]+ } '"'
]],{
  transform = function (t)
    local out = {}
    local added = {}
    for _,v in ipairs(t) do
      if added[v] == nil then
        table.insert(out,string.format('Name: %s\nArgs:\nOutput:\nDescription:\n---',v))
        added[v] = true
      end
    end
    return table.concat(out,'\n')
  end
})

local outfile = io.open('proc_reference.txt','w')
outfile:write(pat:match(sourcet))
outfile:close()
```

**Comment by PaulBatchelor** [comment]
It's a good consideration. 

Documentation in Runt has been low on my list of priorities at the moment. As far as I know, I'm the only one actually using Runt. Even the most mediocre documentation takes a lot of time and effort to do, so I've focused that energy on trying to use the language itself. 

In practice, I've found the approach I took with ugen_reference.txt to be okay, but not worth repeating. I'd probably want to try to build a tool for general documentation that more directly uses the Runt language/parser. Any thoughts or suggestions are welcome.
